src/dialects/mariadb/query-generator.js

```diff
@@ -1,14 +1,57 @@
 'use strict';
 
+import { normalizeDataType } from '../abstract/data-types-utils';
+
 const { MySqlQueryGenerator } = require('../mysql/query-generator');
 const _ = require('lodash');
+const Utils = require('../../utils');
 
 export class MariaDbQueryGenerator extends MySqlQueryGenerator {
 
   _getTechnicalSchemaNames() {
     return ['MYSQL', 'INFORMATION_SCHEMA', 'PERFORMANCE_SCHEMA', 'mysql', 'information_schema', 'performance_schema'];
   }
 
+  addColumnQuery(table, key, dataType, options) {
+    options = options || {};
+
+    const ifNotExists = options.ifNotExists ? ' IF NOT EXISTS' : '';
```

Comment:
I think (without certainty) that joinSQLFragments adds spaces itself

```suggestion
    const ifNotExists = options.ifNotExists ? 'IF NOT EXISTS' : '';
```